src/components/Calculator.js

```diff
@@ -1,43 +1,86 @@
 import React, { Component } from 'react';
+import calculate from './logic/calculate';
 
 // eslint-disable-next-line react/prefer-stateless-function
 class Calculator extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      obj: { next: '', operation: null, total: null },
+    };
+  }
+
+  placeVal = (e) => {
+    this.setState((prevS) => ({
+      obj: calculate(prevS.obj, e.target.name),
+    }));
+  };
+
   render() {
+    const { obj } = this.state;
     return (
       <div className="container">
-        <form>
-          <input className="number-inp" type="number" value="0" />
-        </form>
+        <div className="form">
+          <p>{obj.operation}</p>
+          <input className="number-inp" value={obj.total !== null && obj.next == null ? obj.total : obj.next} readOnly />
+        </div>
         <div className="keys">
-          <button type="button">AC</button>
-          <button type="button">+/-</button>
-          <button type="button">%</button>
-          <button className="orange-btn" type="button">
+          <button name="AC" onClick={this.placeVal} type="button">
+            AC
+          </button>
+          <button name="+/-" onClick={this.placeVal} type="button">
+            +/-
+          </button>
+          <button name="%" onClick={this.placeVal} type="button">
+            %
+          </button>
+          <button name="÷" onClick={this.placeVal} className="orange-btn" type="button">
```

Comment:
- [ ] Kindly create a `Button` component and make it a child of the `Calculator` component.